octavia-cli/octavia_cli/generate/renderers.py

```diff
@@ -161,3 +209,49 @@ def write_yaml(self, project_path: str) -> str:
         with open(output_path, "w") as f:
             f.write(rendered)
         return output_path
+
+
+class ConnectionRenderer(BaseRenderer):
+
+    TEMPLATE = JINJA_ENV.get_template("connection.yaml.j2")
+
+    def __init__(self, connection_name: str, source: resources.Source, destination: resources.Destination) -> None:
+        """Connection renderer constructor.
+
+        Args:
+            connection_name (str): Name of the connection to render.
+            source (resources.Source): Connection's source.
+            destination (resources.Destination): Connections's destination.
+        """
+        super().__init__(connection_name)
+        self.source = source
+        self.destination = destination
+
+    @staticmethod
+    def catalog_to_yaml(catalog: dict) -> str:
+        """Convert the source catalog to a YAML string.
+        Convert camel case to snake case.
+
+        Args:
+            catalog (dict): Source's catalog.
+
+        Returns:
+            str: Catalog rendered as yaml.
+        """
+        return yaml.dump(catalog, Dumper=CatalogDumper, default_flow_style=False)
+
+    def write_yaml(self, project_path: str) -> str:
+        output_path = self._get_output_path(project_path, ConnectionDefinition.type)
+        yaml_catalog = self.catalog_to_yaml(self.source.catalog)
+
+        rendered = self.TEMPLATE.render(
+            {
+                "connection_name": self.resource_name,
+                "source_id": self.source.resource_id,
+                "destination_id": self.destination.resource_id,
+                "catalog": yaml_catalog,
+            }
+        )
+        with open(output_path, "w") as f:
+            f.write(rendered)
+        return output_path
```

Comment:
This function is doing more than writing the yaml file.
For the future maybe break into two: write_yaml file that can be placed in the BaseRenderer class because is used by both sub-classes and other called `build_render` which is an abstract method in BaseRenderer class and need to be implemented for each sub-class